packages/core/src/utils/rpc.ts

```diff
@@ -433,31 +541,64 @@ export const validateLogsAndBlock = (logs: SyncLog[], block: SyncBlock) => {
 export const validateTracesAndBlock = (
   traces: SyncTrace[],
   block: SyncBlock,
+  blockIdentifier: "number" | "hash",
 ) => {
   const traceIds = new Set<string>();
   const transactionHashes = new Set(block.transactions.map((t) => t.hash));
-  for (const trace of traces) {
+  for (const [index, trace] of traces.entries()) {
     if (transactionHashes.has(trace.transactionHash) === false) {
-      throw new Error(
-        `Detected inconsistent RPC responses. 'trace.transactionHash' ${trace.transactionHash} not found in 'block.transactions' ${block.hash}`,
+      const error = new RpcProviderError(
+        `Inconsistent RPC response data. The top-level trace at array index ${index} has a 'transactionHash' of ${trace.transactionHash}, but the associated 'block.transactions' array does not contain a transaction matching that hash.`,
       );
+      error.meta = [
+        "Please report this error to the RPC operator.",
+        eth_getBlockText(
+          blockIdentifier === "number" ? hexToNumber(block.number) : block.hash,
+        ),
+        debug_traceBlockText(
+          blockIdentifier === "number" ? hexToNumber(block.number) : block.hash,
+        ),
+      ];
+      error.stack = undefined;
+      throw error;
     }
 
     const id = `${trace.transactionHash}-${trace.trace.index}`;
     if (traceIds.has(id)) {
-      throw new Error(
-        `Detected invalid debug_traceBlockByHash response. Duplicate trace index ${trace.trace.index} for transaction ${trace.transactionHash}.`,
+      const error = new RpcProviderError(
```

Comment:
This one is a bit strange - if this failed, wouldn't we have introduced it in our DFS post-processing step?